languages/go/exercises/concept/conditionals/.meta/example.go

```diff
@@ -0,0 +1,60 @@
+package _meta
+
+// ParseCard returns the integer value of a card following blackjack ruleset.
+func ParseCard(card string) int {
+	switch card {
+	case "ace":
+		return 11
+	case "two":
+		return 2
+	case "three":
+		return 3
+	case "four":
+		return 4
+	case "five":
+		return 5
+	case "six":
+		return 6
+	case "seven":
+		return 7
+	case "eight":
+		return 8
+	case "nine":
+		return 9
+	case "ten", "jack", "queen", "king":
+		return 10
+	default:
+		return 0
+	}
+}
+
+// Blackjack returns true if the player has a blackjack, false otherwise.
+func Blackjack(card1, card2 string) bool {
+	return ParseCard(card1)+ParseCard(card2) == 21
+}
+
+// FirstTurn returns the "optimal" decision for the first turn, given the cards of the player and the dealer.
+func FirstTurn(card1, card2, dealerCard string) string {
+	handScore := ParseCard(card1) + ParseCard(card2)
+	// Single if statement
+	if handScore == 22 {
+		return "P"
+	}
+	dealerScore := ParseCard(dealerCard)
+	// Nested if statement (no else)
+	if Blackjack(card1, card2) {
+		if dealerScore < 10 {
+			return "W"
+		}
+		return "S"
+	}
+	// Switch statement
+	switch {
+	case (handScore >= 17) || (handScore >= 12 && dealerScore <= 6):
+		return "S"
+	case (handScore <= 11) || (handScore >= 12 && dealerScore > 6):
+		return "H"
+	default:
+		return ""
+	}
+}
```

Comment:
Wondering if this still gives the student too much freedom to do unidiomatic things which would be hard to check automatically. (Since we want 95% solutions of the concept exercises be handled automatically.)

My suggestion:

Split this in more functions:
```go
func FirstTurn(card1, card2, dealerCard string) string {
	handScore := ParseCard(card1) + ParseCard(card2)
	dealerScore := ParseCard(dealerCard)
	isBlackjack := Blackjack(card1, card2)

	s := largeHand(handScore, isBlackjack, dealerScore)
	if s != "" {
		return s
	}
	return smallHand(handScore, dealerScore)
}

func largeHand(handScore int, isBlackjack bool, dealerScore int) string {
	// Single if statement
	if handScore == 22 {
		return "P"
	}
	// Nested if statement (no else)
	if isBlackjack {
		if dealerScore < 10 {
			return "W"
		}
		return "S"
	}
	return ""
}

func smallHand(handScore int, dealerScore int) string {
	// Switch statement
	switch {
	case (handScore >= 17) || (handScore >= 12 && dealerScore <= 6):
		return "S"
	case (handScore <= 11) || (handScore >= 12 && dealerScore > 6):
		return "H"
	default:
		return ""
	}
}
```

`FirstTurn` would be part of the stub and the student would not have to implement it. The other 2 functions would need to be implemented by the student.

What do you think? @andres-zartab 

@ErikSchierboom: What do you think? Or is this ok as it is?

Reply:
I am wondering if we could somehow simplify things a bit. There is quite a bit of code involved, and I'm wondering if this would lead to many different representations (which would make commenting on those representations less efficient). I like the `ParseCard` and `Blackjack` functions. Maybe we could simplify or replace the `FirstTurn` function a bit?

Reply:
I really like the `smallHand` function I mentioned above as a function the student could implement.

Reply:
Will give some thought to this now and propose something shortly